samples/rocketmq-spring-boot-starter-sample/rocketmq-produce-demo/src/main/java/org/apache/rocketmq/samples/springboot/ProducerApplication.java

```diff
@@ -111,53 +111,54 @@ private void testTransaction() throws MQClientException {
     }
 
     @RocketMQTransactionListener(txProducerGroup = TX_PGROUP_NAME)
-    class TransactionListenerImpl implements TransactionListener {
+    class TransactionListenerImpl implements RocketMQLocalTransactionListener {
         private AtomicInteger transactionIndex = new AtomicInteger(0);
 
         private ConcurrentHashMap<String, Integer> localTrans = new ConcurrentHashMap<String, Integer>();
 
         @Override
-        public LocalTransactionState executeLocalTransaction(Message msg, Object arg) {
-            System.out.printf("------ executeLocalTransaction is executed, msgTransactionId=%s %n", msg.getTransactionId());
+        public RocketMQLocalTransactionState executeLocalTransaction(Message msg, Object arg) {
+            System.out.printf("------ executeLocalTransaction is executed, msgTransactionId=%s %n",
+                MessageHelper.getTransactionId(msg));
             int value = transactionIndex.getAndIncrement();
             int status = value % 3;
-            localTrans.put(msg.getTransactionId(), status);
+            localTrans.put(MessageHelper.getTransactionId(msg), status);
             if (value == 0) {
                 // Return local transaction with success(commit), in this case,
                 // this message will not be checked in checkLocalTransaction()
-                System.out.printf("------ Simulating msg %s related local transaction exec succeeded! %n", new String(msg.getBody()));
-                return LocalTransactionState.COMMIT_MESSAGE;
+                System.out.printf("------ Simulating msg %s related local transaction exec succeeded! %n", msg.getPayload());
+                return RocketMQLocalTransactionState.COMMIT_MESSAGE;
             }
 
             if (value == 1) {
                 // Return local transaction with failure(rollback) , in this case,
                 // this message will not be checked in checkLocalTransaction()
-                System.out.printf("------ Simulating %s related local transaction exec failed! %n", new String(msg.getBody()));
-                return LocalTransactionState.ROLLBACK_MESSAGE;
+                System.out.printf("------ Simulating %s related local transaction exec failed! %n", msg.getPayload());
+                return RocketMQLocalTransactionState.ROLLBACK_MESSAGE;
             }
 
-            return LocalTransactionState.UNKNOW;
+            return RocketMQLocalTransactionState.UNKNOW;
         }
 
         @Override
-        public LocalTransactionState checkLocalTransaction(MessageExt msg) {
-            LocalTransactionState retState = LocalTransactionState.COMMIT_MESSAGE;
-            Integer status = localTrans.get(msg.getTransactionId());
+        public RocketMQLocalTransactionState checkLocalTransaction(Message msg) {
+            RocketMQLocalTransactionState retState = RocketMQLocalTransactionState.COMMIT_MESSAGE;
```

Comment:
the initial state is RocketMQLocalTransactionState.COMMIT_MESSAGE,
Why not RocketMQLocalTransactionState. UNKNOW?

Reply:
you are right, I will change it as UKNOW.